kmip/services/server/session.py

```diff
@@ -51,6 +51,9 @@ def __init__(self, engine, connection, name=None):
             kwargs={}
         )
 
+        if name is None:
+            name = threading.current_thread().name
```

Comment:
Yes! There are a couple of places you can go to discuss PyKMIP. We have an IRC channel, #pykmip, on Freenode, hosted at https://webchat.freenode.net/. Login is free and registration is optional, if you haven't tried it before. We try to hang out there on the channel on a regular basis.

You can also send me or @rellerreller email with any questions you may have. This project isn't too active, and there aren't many people working on it, so we don't have a dedicated mailing list for it yet.

As for your questions, the `KmipSession` is a thread object that will be spun off from a new `KmipServer` class that I am finishing up right now; it's all part of the main KMIP server rewrite I've been working on for the past few months. The `KmipServer` handles receiving client connections and spins off `KmipSessions` for each one. The session then parses the incoming request and (if parse successful) passes the resulting KMIP request object to the `KmipEngine` for processing.

Since the `KmipServer` isn't out yet, there's no direct way to invoke `KmipSessions` yet without creating your own framework for it, which it sounds like you're doing with the SocketServer/AsynchronousMixins. It is definitely possible to run a `KmipSession` directly as your main thread; instead of invoking `KmipSession.start` (which kicks off the separate threaded-ness of the session) just call `KmipSession.run` directly.

The reason I wanted to avoid using the current thread's name is because the intent is to have multiple `KmipSessions` at once and I wanted to avoid any thread naming collisions. While I don't think there's anything technically incorrect about that thread naming scheme, thread activity is logged by thread name and I wanted to make it easy for those looking through the logs to distinguish between different threads.

Does that answer your questions? Definitely let me know if you have more. I'm happy to help.